aws/resource_aws_rds_cluster.go

```diff
@@ -712,25 +712,34 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 		}
 
 	} else {
-		if _, ok := d.GetOk("master_password"); !ok {
-			return fmt.Errorf(`provider.aws: aws_rds_cluster: %s: "master_password": required field is not set`, d.Get("database_name").(string))
-		}
 
-		if _, ok := d.GetOk("master_username"); !ok {
-			return fmt.Errorf(`provider.aws: aws_rds_cluster: %s: "master_username": required field is not set`, d.Get("database_name").(string))
+		if _, ok := d.GetOk("global_cluster_identifier"); !ok {
+			if _, ok := d.GetOk("master_password"); !ok {
+				return fmt.Errorf(`provider.aws: aws_db_instance: %s: "master_password": required field is not set`, d.Get("name").(string))
```

Comment:
`name` is not an attribute in this resource's schema, so can cause a panic as found by the acceptance testing:

```
=== CONT  TestAccAWSRDSCluster_missingUserNameCausesError

------- Stderr: -------
panic: interface conversion: interface {} is nil, not string

goroutine 137 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsRDSClusterCreate(0xc000349260, 0x3c28320, 0xc0001adc00, 0xc000349260, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_rds_cluster.go:718 +0x6d9d
```

Will fix this on merge.